packages/provider-utils/src/response-handler.ts

```diff
@@ -184,3 +184,57 @@ export const createJsonResponseHandler =
       rawValue: parsedResult.rawValue,
     };
   };
+
+export const createBinaryResponseHandler =
+  (): ResponseHandler<ArrayBuffer> =>
```

Comment:
ResponseHandler<UInt8Array> ?

Reply:
I played with this -- if I omit the generic, I get `Generic type 'ResponseHandler' requires 1 type argument(s).` We can omit the return type as it's inferrable, but then we get `Binding element 'response' implicitly has an 'any' type` on the
de-sugared args on line 190, which we could probably fix by specifying their types but that's more verbose than just keeping line 189.

I may be missing what you're getting at, happy to look at this further later.